Tests/Unit/Components/CategoryResolver/DefaultCategoryResolverTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * (c) shopware AG <info@shopware.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ShopwarePlugins\Connect\Tests\Unit\Components\CategoryResolver;
+
+use Shopware\Components\Model\ModelManager;
+use Shopware\CustomModels\Connect\ProductToRemoteCategoryRepository;
+use Shopware\CustomModels\Connect\RemoteCategoryRepository;
+use ShopwarePlugins\Connect\Components\CategoryResolver\DefaultCategoryResolver;
+use ShopwarePlugins\Connect\Tests\AbstractConnectUnitTest;
+
+class DefaultCategoryResolverTest extends AbstractConnectUnitTest
+{
+    /**
+     * @var DefaultCategoryResolver
+     */
+    private $defaultCategoryResolver;
+
+    /**
+     * @var  RemoteCategoryRepository
+     */
+    private $remoteCategoryRepository;
+
+    /**
+     * @var ModelManager
+     */
+    private $manager;
+
+    /**
+     * @before
+     */
+    public function prepareMocks()
+    {
+        $this->remoteCategoryRepository = $this->createMock(RemoteCategoryRepository::class);
+        $this->manager = $this->createMock(ModelManager::class);
+
+        $this->defaultCategoryResolver = new DefaultCategoryResolver(
+            $this->manager,
+            $this->remoteCategoryRepository,
+            $this->createMock(ProductToRemoteCategoryRepository::class)
+        );
+    }
+
+    public function test_store_remote_categories_success()
+    {
+        $articleId = 4;
+        $categories = ['Category 1'];
```

Comment:
Would be better to use ```$categories = ['category1' => 'Category 1'];``` instead of ```$categories = ['Category 1'];```, because we use this structure in Product::categories.
Btw your test handles only 1 case. 

@svenfinke what do you think ?

Reply:
I will add some fails etc to the tests and fix that category thing